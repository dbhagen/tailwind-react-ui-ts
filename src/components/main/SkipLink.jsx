import React from 'react'
import PropTypes from 'prop-types'

import { Box } from '../primitives'
import { FillButton } from '../button'

const SkipLink = ({ children, href, ...rest }) => (
  <Box visuallyHiddenFocusable>
    <FillButton
      is="a"
      bg="white"
      text="black"
      absolute
      rounded="none"
      href={href}
      {...rest}
    >
      {children || 'Skip to main content'}
    </FillButton>
  </Box>
)

SkipLink.propTypes = {
  children: PropTypes.node,
  href: PropTypes.string,
}

SkipLink.defaultProps = {
  children: undefined,
  href: '#main',
}

export default SkipLink
