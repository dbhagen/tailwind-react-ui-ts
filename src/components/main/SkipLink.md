Styled button that shows on focus.
