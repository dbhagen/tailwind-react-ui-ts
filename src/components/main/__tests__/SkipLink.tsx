import React from 'react'
import { shallow } from 'enzyme'

import SkipLink from '../SkipLink'

const setup = (testProps = {}) => {
  const props = Object.assign(testProps)

  const wrapper = shallow(<SkipLink {...props} />)

  return {
    props,
    wrapper,
  }
}

describe('SkipLink', () => {
  it('renders matching snapshot', () => {
    const { wrapper } = setup()

    expect(wrapper).toMatchSnapshot()
  })
})
