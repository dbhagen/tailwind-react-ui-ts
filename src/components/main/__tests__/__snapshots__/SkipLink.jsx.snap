// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SkipLink renders matching snapshot 1`] = `
<Box
  inline={false}
  inlineBlock={false}
  is="div"
  visuallyHiddenFocusable={true}
>
  <FillButton
    absolute={true}
    bg="white"
    href="#main"
    is="a"
    rounded="none"
    text="black"
  >
    Skip to main content
  </FillButton>
</Box>
`;
